soda/scientific/soda/scientific/distribution/comparison.py

```diff
@@ -35,50 +36,60 @@ class MissingCategories(Exception):
     """Thrown when a category in the test data is missing from the ref data."""
 
 
+class DistributionRefIncompatibleException(Exception):
+    """Thrown when the DRO dtype is incompatible with the test that is used."""
+
+
 class DistributionChecker:
     def __init__(self, distribution_check_cfg: DistributionCheckCfg, data: List[Any]):
         cfg = DistCfg(
             reference_file_path=distribution_check_cfg.reference_file_path,
         )
         self.test_data = data
 
-        self.ref_cfg = self._parse_reference_cfg(cfg.reference_file_path)
+        algo_mapping = {"chi_square": ChiSqAlgorithm, "ks": KSAlgorithm, "swd": SWDAlgorithm, "psi": PSIAlgorithm}
+        self.method = distribution_check_cfg.method
+        self.choosen_algo = algo_mapping[self.method]
 
-        algo_mapping = {
-            "categorical": ChiSqAlgorithm,
-            "continuous": KSAlgorithm,
-            "chi_square": ChiSqAlgorithm,
-            "ks": KSAlgorithm,
-        }
-        self.choosen_algo = algo_mapping[self.ref_cfg.method]
+        self.ref_cfg = self._parse_reference_cfg(cfg.reference_file_path)
 
     def run(self) -> Tuple[float, float]:
         test_data = pd.Series(self.test_data)
 
         bootstrap_size = 10
+        check_values = []
         stat_values = []
-        p_values = []
 
         for i in range(bootstrap_size):
-            stat_value, p_value = self.choosen_algo(self.ref_cfg, test_data, seed=i).evaluate()
-            stat_values.append(stat_value)
-            p_values.append(p_value)
+            check_results = self.choosen_algo(self.ref_cfg, test_data, seed=i).evaluate()
+
+            check_value = check_results.get("check_value")
+            check_values.append(check_value)
 
-        stat_value = np.median(stat_values)
-        p_value = np.median(p_values)
+            stat_value = check_results.get("stat_value")
+            if stat_value:
+                stat_values.append(stat_value)
 
-        return stat_value, p_value
+        check_value = np.median(check_values)
+        if stat_values:
+            stat_value = np.median(stat_values)
+
+        return dict(stat_value=stat_value, check_value=check_value)
 
     def _parse_reference_cfg(self, ref_file_path: FilePath) -> RefDataCfg:
         with open(str(ref_file_path)) as stream:
             try:
                 parsed_file: dict = yaml.safe_load(stream)
                 ref_data_cfg = {}
-                if "method" in parsed_file:
-                    ref_data_cfg["method"] = parsed_file["method"]
+
+                correct_configs = {"continuous": ["ks", "psi", "swd"], "categorical": ["chi_square", "psi", "swd"]}
```

Comment:
The Wasserstein metric is also know as Earth Mover's distance https://en.wikipedia.org/wiki/Earth_mover%27s_distance

We might want to allow for `emd` as an option here and when we write docs refer to both as one and the same and say they can be used interchangeably